pkgs/development/compilers/openjdk/9.nix

```diff
@@ -0,0 +1,255 @@
+             sha256 = "13rqfw214wrskfbjjg02grvi4hdr27g8idfdg00hvcfgak1s1i6j";
+          };
+  jdk = fetchurl {
+             url = "${baseurl}/jdk/archive/${repover}.tar.gz";
+             sha256 = "172xf7q21r1jblbayaqg2kk074nphycby0hmgbfiyg7lqh2w0dqq";
+          };
+  jaxws = fetchurl {
+             url = "${baseurl}/jaxws/archive/${repover}.tar.gz";
+             sha256 = "1i55pzh2g5lp3injj2vrm72ipl6jrf0kjl5c53f98fg86095ap1d";
+          };
+  jaxp = fetchurl {
+             url = "${baseurl}/jaxp/archive/${repover}.tar.gz";
+             sha256 = "0031ch90kdxnzpnm9jizwcabgsxylaqmchwj11gc4f89jcp49amw";
+          };
+  nashorn = fetchurl {
+             url = "${baseurl}/nashorn/archive/${repover}.tar.gz";
+             sha256 = "15qswf7agjshpz6g00gbapwlrbc2n7mawnwam6sg02cdmbmw5grg";
+          };
+  openjdk9 = stdenv.mkDerivation {
+    # name = "openjdk-9u${update}b${build}";
+    name = "openjdk-9ea-b${build}";
+
+    srcs = [ jdk9 langtools hotspot corba jdk jaxws jaxp nashorn ];
+    sourceRoot = ".";
+
+    outputs = [ "out" "jre" ];
+
+    nativeBuildInputs = [ pkgconfig ];
+    buildInputs = [
+      cpio file which unzip zip perl bootjdk zlib cups freetype alsaLib
+      libjpeg giflib libX11 libICE libXext libXrender libXtst libXt libXtst
+      libXi libXinerama libXcursor lndir fontconfig
+    ] ++ lib.optionals (!minimal && enableGnome2) [
+      gtk2 gnome_vfs GConf glib
+    ];
+
+    prePatch = ''
+      ls | grep jdk | grep -v '^jdk9' | awk -F- '{print $1}' | while read p; do
+        mv $p-* $(ls | grep '^jdk9')/$p
+      done
+      cd $(ls | grep '^jdk9')
+    '';
+
+    patches = [
+      ./fix-java-home-jdk9.patch
+      ./read-truststore-from-env-jdk9.patch
+      ./currency-date-range-jdk8.patch
+    #] ++ lib.optionals (!minimal && enableInfinality) [
+    #  ./004_add-fontconfig.patch
+    #  ./005_enable-infinality.patch
+    ] ++ lib.optionals (!minimal && enableGnome2) [
+      ./swing-use-gtk-jdk9.patch
+    ];
+
+    preConfigure = ''
+      chmod +x configure
+      substituteInPlace configure --replace /bin/bash "$shell"
+      # substituteInPlace hotspot/make/linux/adlc_updater --replace /bin/sh "$shell"
+
+      configureFlagsArray=(
+        "--with-boot-jdk=${bootjdk.home}"
+        "--with-update-version=${update}"
+        "--with-build-number=${build}"
+        "--with-milestone=fcs"
+        "--enable-unlimited-crypto"
+        "--disable-debug-symbols"
+        "--disable-freetype-bundling"
+        "--with-zlib=system"
+        "--with-giflib=system"
+        "--with-stdc++lib=dynamic"
+
+        # glibc 2.24 deprecated readdir_r so we need this
+        # See https://www.mail-archive.com/openembedded-devel@lists.openembedded.org/msg49006.html
+        "--with-extra-cflags=-Wno-error=deprecated-declarations -Wno-error=format-contains-nul -Wno-error=unused-result"
+    ''
+    + lib.optionalString minimal "\"--disable-headful\""
+    + ");";
+
+    NIX_LDFLAGS= lib.optionals (!minimal) [
```

Comment:
Please use `pkg-config --libs` for all these libs such that it will continue to work.